JL.Windows/GUI/PopupWindow.xaml.cs

```diff
@@ -695,6 +695,31 @@ private StackPanel PrepareResultStackPanel(LookupResult result, int index, int r
             audioButton.Click += AudioButton_Click;
 
             _ = top.Children.Add(audioButton);
+
+            if (!configManager.MineToFileInsteadOfAnki)
+            {
+                if (MiningUtils.CheckDuplicate(result).Result)
```

Comment:
Accessing the `Result` property of a `Task` is a blocking operation, thus it should be avoided. Instead we should do something like

```C#
bool duplicateCard = await MiningUtils.CheckDuplicate(result).ConfigureAwait(true);
if (duplicateCard)
```

Reply:
This is the biggest thing that I'm having trouble with since this method is synchronous, toying with something to do the check for duplicates after the window is loaded and toggling the visibility so there is 0 impact on loading the window.